server.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const url = require('url');
+const querystring = require('querystring');
+const http = require('http');
+const parseBody = require('./lib/parse.js');
+const badRequest = require('./lib/bad-request.js');
+
+const cowsay = require('cowsay');
+
+const PORT = process.env.PORT || 3000;
+const server = http.createServer(function(req, res){
+  req.url = url.parse(req.url);
+  req.url.query = querystring.parse(req.url.query);
+
+  if(req.method === 'GET' && req.url.pathname === '/api/cowsay'){
+    // console.log(req.url, 'req.url');
+    // console.log(req.url.query, 'req.url.query');
+    if(!req.url.query.text){
+      return badRequest(res);
+    }
+    res.statusCode = 200;
+    res.setHeader('Content-Type', ' text/html');
+    res.write(generateCowSay(req.url.query.text));
+    res.end();
+    return;
+  }
+
+  function generateCowSay(text){
```

Comment:
Not a point-loser, but this function should be modularized in its own module in the lib folder, and then required in the server module.